yaguzmang/ghg-emission-calculation-tool
# Feature/upgrade next js to app directory

Closes #39 

## Description


- Add a prettier pluggin to order tailwind classes automatically.
- Update frontend dependencies.
- Upgrade to Next.js 13.4 with App directory, as it is now production ready. See: [Next.js 13.4](https://nextjs.org/blog/next-13-4).

Recommend steps to run the app after pulling the changes:

- In the frontend folder: Delete your local `.next`, `node_modules`, directories, and the `package-lock.json` file (sometimes this one causes errors).
- Run `npm install` (using the latest version if possible)

## How to test

Steps to test:
- New App directory:
  - [x] Run: `npm run dev`
  - [x] Check that the app renders ‘Tailwind Test’ with a green background in your localhost/. 
- Formatting:
  - [x] In the  page.tsx  file, in the `<main>` tag add the class name `text-lg` before `bg-green-500`. 
  Run `npm run format:write` and check that the order becomes `bg-green-500 text-lg`.
- Pre-commit hook (after changes to accommodate to the App directory): 
    - [ ] introduce an error (you can delete a closing tag in the page.tsx file for example). Stage the file using `git add app/page.tsx`. 
    - [x] Run `git commit` and the commit should be aborted and you should get an error.
    - [x] Discard the changes made.
    - [x] Introduce a formatting mistake (you can delete just a ‘;’ in the page.tsx file for example). 
    Stage the file using `git add app/page.tsx`. 
    - [x] Run `git commit` and you should be prompted to write a commit message. Check that in your staged file the format mistake has been overwritten (the ‘;’ should now be there).
    - [x] Abort the commit.
